src/lib/utils/notifications/group.ts

```diff
@@ -0,0 +1,206 @@
+import { getAuthorName } from "$lib/utils/client/member";
+import {
+  NotificationType,
+  SHOULD_MERGE_NOTIFICATIONS,
+} from "$lib/utils/notifications/types";
+import type {
+  Author,
+  CustomAuthor,
+  Notification as DBNotification,
+  Mandate,
+  Member,
+  Position,
+} from "@prisma/client";
+
+// A notification as it is returned from prisma query
+type Notification = DBNotification & {
+  fromAuthor:
+    | (Author & {
+        member: Member;
+        mandate:
+          | (Mandate & {
+              position: Position;
+            })
+          | null;
+        customAuthor: CustomAuthor | null;
+      })
+    | null;
+};
```

Comment:
If we can use [this pattern instead](https://www.prisma.io/docs/orm/prisma-client/type-safety/operating-against-partial-structures-of-model-types#solution-1), I'd prefer that.

Reply:
Aww nice, was looking for something like that!